### PR TITLE
Add role and group information in transformations

### PR DESCRIPTION
**Pull request by AkselMeola**
INSLYPROD-6519

### CODE REVIEW

File: src/Api/Transformers/Master/UserTransformer.php [hunk 1]
```diff
@@ -14,5 +14,10 @@ class UserTransformer extends AbstractTransformer
         'role_id' => 'roleId',
         'broker_id' => 'brokerId',
         'insly3_oid' => 'insly3Oid',
+        'role' => 'role',
```
Comment (taron-insly) [inline src/Api/Transformers/Master/UserTransformer.php:17]: Such as role model and transformer exists you can add relation already:

```php
    public function relations(): array
    {
        return [
            'role' => Relation::create(
                Relation::TO_ONE,
                Role::class,
                RoleTransformer::class
            )
        ];
    }
```

File: src/Api/Models/User.php [hunk 2]
```diff
@@ -15,4 +15,7 @@ class User extends Model
     public string $insly3Oid;
 
     public array $props = [];
+
+    // TODO: Refactor to use relations
+    public array $role = [];
```
Comment (taron-insly) [inline src/Api/Models/User.php:20]:
```suggestion
    public Role $role;
```

**Review by taron-insly — CHANGES_REQUESTED** [review]

File: src/Api/Transformers/Master/UserTransformer.php (hunk 1 above)
Comment (AkselMeola) [reply src/Api/Transformers/Master/UserTransformer.php:17]: Yes for new endpoints/transformers. This is legacy endpoint. I'm sorry but in this timeframe I'm unable to refactor those endppoints.

File: src/Api/Models/User.php (hunk 2 above)
Comment (AkselMeola) [reply src/Api/Models/User.php:20]: Since it is not transformed yet this must remain array.

Comment (AkselMeola) [reply src/Api/Models/User.php]: Added. But Role might be not set on user so it is `?Role`

**Review by taron-insly — APPROVED** [review]

**Pull request merged**